internal/services/network/virtual_hub_resource.go

```diff
@@ -167,12 +175,14 @@ func resourceVirtualHubCreateUpdate(d *pluginsdk.ResourceData, meta interface{})
 	t := d.Get("tags").(map[string]interface{})
 
 	hubRoutingPreference := d.Get("hub_routing_preference").(string)
+	autoscale := expandAutoScaleConfig(d)
 
 	parameters := network.VirtualHub{
 		Location: utils.String(location),
 		VirtualHubProperties: &network.VirtualHubProperties{
-			RouteTable:           expandVirtualHubRoute(route),
-			HubRoutingPreference: network.HubRoutingPreference(hubRoutingPreference),
+			RouteTable:                          expandVirtualHubRoute(route),
+			HubRoutingPreference:                network.HubRoutingPreference(hubRoutingPreference),
+			VirtualRouterAutoScaleConfiguration: autoscale,
```

Comment:
we could avoid using the expand function now that we've flattened the property and set the autoscaleconfig here instead 

```
	parameters := network.VirtualHub{
		Location: utils.String(location),
		VirtualHubProperties: &network.VirtualHubProperties{
			RouteTable:           expandVirtualHubRoute(route),
			HubRoutingPreference: network.HubRoutingPreference(hubRoutingPreference),
		},
		Tags: tags.Expand(t),
	}

	if v, ok := d.GetOk("virtual_router_auto_scale_min_capacity"); ok {
		parameters.VirtualHubProperties.VirtualRouterAutoScaleConfiguration = &network.VirtualRouterAutoScaleConfiguration{
			MinCapacity: utils.Int32(int32(v.(int))),
		}
	}
```